task3/UI/scripts/PostView.js

```diff
@@ -0,0 +1,101 @@
+class PostView {
+    isGuest = true;
+    userName;
+    feedPosts = document.querySelector('div.section__data');
+    _postTemplate = document.getElementById('post__template');
+
+    constructor(isGuest, userName) {
+        this.isGuest = isGuest;
+        if (!isGuest) {
+            this.userName = userName;
+        }
+    }
+
+    drawPost(postData) {
+        let newPost = document.importNode(this._postTemplate.content, true);
+        this._fillPostData(newPost, postData);
+        let article = newPost.querySelector('div.post');
+        if (this.isGuest) {
+            article.appendChild(this._drawLike());
+        } else if (this.userName === postData.author) {
+            this._drawPostButtons();
+        }
+        this.feedPosts.insertBefore(newPost, this.feedPosts.firstChild);
+    }
+
+    redrawPost(id, newPostData) {
+        let oldPost = document.getElementById(id);
+        let newPost = document.importNode(this._postTemplate.content, true);
+        this._fillPostData(newPost, newPostData);
+        let article = newPost.querySelector("div.post");
+        if (!this.isGuest) {
+            article.appendChild(this._drawLike());
+        } else if (this.userName === newPost.author) {
+            article.appendChild(this._drawPostButtons());
+        }
+        this.feedPosts.replaceChild(newPost, oldPost);
+    }
+
+    deletePost(id) {
+        let post = document.getElementById(id);
+        this.feedPosts.removeChild(post);
+    }
+
+    _fillPostData(newPost, postData) {
+        console.log(newPost);
```

Comment:
I think it might be removed

Reply:
Oops, forgot to delete 😅

Reply:
I fixed all the bugs, thanks again for your help.
Can I get the final grade for this task?)

Reply:
After I log in I can see 'add post' button, but log in button is still here:
![Screen Shot 2021-05-05 at 3 54 52 PM](https://user-images.githubusercontent.com/22647154/117144181-4f3f4180-adba-11eb-89c2-65a71bb6e3d4.png)

That's why grade is 9, but again you still can fix it.

Reply:
I apologize, my fault, now everything should work here (if not, then it definitely works in task 10 😄)
Thank you)